lib/fs.js

```diff
@@ -95,7 +98,15 @@ function sanitizeFD(fd) {
 // for callbacks that are passed to the binding layer, callbacks that are
 // invoked from JS already run in the proper scope.
 function makeCallback(cb) {
-  if (typeof cb !== 'function') {
+  if (typeof cb === 'undefined') {
+    // TODO(thefourtheye): throw error instead of deprecation warning in v8.0.0
+    // Refer the discussion: https://github.com/nodejs/node/pull/7846
+    let m = 'async function calls without valid callbacks are deprecated.';
+    if (!process.traceDeprecation)
+      m += ' Use --trace-deprecation to find the function responsible.';
```

Comment:
I'm not sure this additional part about `--trace-deprecation` is necessary but can live with it.